sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -356,90 +364,39 @@ func (host *pythonLanguageHost) Pack(ctx context.Context, req *pulumirpc.PackReq
 	}, nil
 }
 
-// prepareVirtualEnvironment will create and install dependencies in the virtual environment if host.virtualenv is set.
-func (host *pythonLanguageHost) prepareVirtualEnvironment(ctx context.Context, root, cwd, virtualenv string) error {
```

Comment:
Pip specific logic was moved to EnsureVenv in pip.go, we only retain the creation of the engine log message writers here.